de-lib/src/main/java/org/iplantc/de/client/models/UserSettings.java

```diff
@@ -286,6 +290,23 @@ public void useDefaultValues(UserInfo instance) {
         setValues(defaults);
     }
 
+    String getHomeDir(UserInfo userInfo) {
+        String userHomePath = userInfo.getHomePath();
+
+        if (!Strings.isNullOrEmpty(userHomePath)) {
+            return userHomePath;
+        }
+
+        String irodsHome = DEProperties.getInstance().getIrodsHomePath();
+        String username = userInfo.getUsername();
+
+        if (Strings.isNullOrEmpty(irodsHome) || Strings.isNullOrEmpty(username)) {
+            return "";
+        }
+
+        return irodsHome + "/" + username;
```

Comment:
I'm thinking this logic should be moved into `UserInfo#getHomePath` itself.

Notice how many other places this method is called besides the `DEFAULT_OUTPUT_FOLDER` default setting, so those parts of the DE will need this default path as well if the bootstrap endpoint couldn't return the real home path.

Same goes for `getTrashPath` and `getBaseTrashPath`. Then when the work for CORE-8367 is done for the data-info serivce, we'll need a way to fetch these real paths once the service comes back up (I'm thinking of adding a "data-info bootstrap" endpoint to do this for CORE-7813).